cirq/google/engine/engine_processor.py

```diff
@@ -108,6 +110,21 @@ def get_device_specification(
         else:
             return None
 
+    def get_device(
+            self,
+            gate_sets: Iterable[serializable_gate_set.SerializableGateSet]):
```

Comment:
Just a side comment: This probably really shows that the gate set should also be part of the device spec.

Reply:
Perhaps there is a better way of doing it, but the gate set is part of the device spec.  You just need a way to turn 'fsim' into 'cirq.FSimGate'.  There needs to be a place that can translate the gate ids into cirq objects.